samples/tfm_integration/psa_crypto/src/psa_crypto.c

```diff
@@ -191,6 +208,325 @@ static psa_status_t crp_gen_key_secp256r1(psa_key_id_t key_id,
+
+	status = al_psa_status(psa_crypto_init(), __func__);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("Crypto init failed.");
+		goto err;
+	}
+
+	LOG_INF("PSA crypto init completed");
+	al_dump_log();
+
+	/* prime256v1 (ecdsa-with-SHA256) private key. */
+#if CONFIG_PSA_IMPORT_KEY
+#if CONFIG_PRIVATE_KEY_STATIC
+	/* This value is based on the private key in user.pem,
+	 * which can be viewed viw the following command:
+	 *
+	 *   $ openssl ec -in user.pem -text -noout
+	 */
+	uint8_t priv_key_data[32] = {
+		0x14, 0xbc, 0xb9, 0x53, 0xa4, 0xee, 0xed, 0x50,
+		0x09, 0x36, 0x92, 0x07, 0x1d, 0xdb, 0x24, 0x2c,
+		0xef, 0xf9, 0x57, 0x92, 0x40, 0x4f, 0x49, 0xaa,
+		0xd0, 0x7c, 0x5b, 0x3f, 0x26, 0xa7, 0x80, 0x48
+	};
+#else /* !CONFIG_PRIVATE_KEY_STATIC */
+	/* Randomly generate the private key. */
+	uint8_t priv_key_data[32] = { 0 };
+
+	LOG_INF("Generate rnadom data for private key");
+	al_dump_log();
+
+	psa_generate_random(priv_key_data, sizeof(priv_key_data));
+	LOG_INF("Random data generation for private key completed");
+	al_dump_log();
+
+#endif /* CONFIG_PRIVATE_KEY_STATIC */
+
+	/* Generate persistent prime256v1 (ecdsa-with-SHA256) key w/ID #1. */
+	/* PSA_KEY_USAGE_EXPORT can be added for debug purposes. */
+	status = al_psa_status(
+		crp_imp_key_secp256r1(key_slot,
+				      PSA_KEY_USAGE_SIGN_HASH |
+				      PSA_KEY_USAGE_VERIFY_HASH,
+				      priv_key_data),
+		__func__);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("Failed to create persistent key #%d", key_slot);
+		goto err;
+	}
+#else /* !CONFIG_PSA_IMPORT_KEY */
+
+	/* NOTE: The certificate signing request (CSR) can be generated using
+	 * openssl by using following commands:
+	 *
+	 * Generate a new key:
+	 *
+	 *   $ openssl ecparam -name secp256k1 -genkey -out USER.key
+	 *
+	 * Generate a certificate signing request, containing the user public key
+	 * and required details to be inserted into the user certificate.
+	 * openssl req -new -key USER.key -out USER.csr \
+	 *   -subj "/O=Linaro/CN=$(uuidgen | tr '[:upper:]' '[:lower:]')"
+	 *
+	 */
+
+	/* Generate persistent prime256v1 (ecdsa-with-SHA256) key w/ID #1. */
+	/* PSA_KEY_USAGE_EXPORT can be added for debug purposes. */
+	status = al_psa_status(
+		crp_gen_key_secp256r1(key_slot,
+				      PSA_KEY_USAGE_SIGN_HASH |
+				      PSA_KEY_USAGE_VERIFY_HASH),
+		__func__);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("Failed to create persistent key #%d", key_slot);
+		goto err;
+	}
+#endif /* CONFIG_PSA_IMPORT_KEY */
+
+	status = al_psa_status(
+		psa_open_key(key_slot, &key_handle),
+		__func__);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("Failed to open persistent key #%d", key_slot);
+		goto err;
+	}
+
+	psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+
+	psa_get_key_attributes(key_handle, &attributes);
+	mbedtls_x509write_csr_set_md_alg(&req, MBEDTLS_MD_SHA256);
+
+	LOG_INF("Adding subject name to CSR");
+	al_dump_log();
+
+	status = mbedtls_x509write_csr_set_subject_name(&req, "O=Linaro,CN=Device Certificate");
```

Comment:
This will need to be device-unique before real use, but is ok for now.

Reply:
@urutva What we do with the test CA we've been using is to append a UUID here, which we could add to the sample.

For example, with `openssl` we do something like this to test:

```bash
$ openssl req -new -key USER.key -out USER.csr \
    -subj "/O=localhost/CN=$(uuidgen | tr '[:upper:]' '[:lower:]')"
```

Reply:
We already generate random data from TF-M as part of this demo ... perhaps we could extract 128 bits from that stream, setting specific bits for a valid UUID as per RFC 4122, described here: https://www.cryptosys.net/pki/uuid-rfc4122.html

```
 The procedure to generate a version 4 UUID is as follows:

    Generate 16 random bytes (=128 bits)
    Adjust certain bits according to RFC 4122 section 4.4 as follows:
        set the four most significant bits of the 7th byte to 0100'B, so the high nibble is "4"
        set the two most significant bits of the 9th byte to 10'B, so the high nibble will be one of "8", "9", "A", or "B" (see Note 1).
    Encode the adjusted bytes as 32 hexadecimal digits
    Add four hyphen "-" characters to obtain blocks of 8, 4, 4, 4 and 12 hex digits
    Output the resulting 36-character string "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
```